src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -10449,6 +10570,44 @@ public void Main(string? s)
                 );
         }
 
+        [Fact]
+        [WorkItem(29916, "https://github.com/dotnet/roslyn/issues/29916")]
+        public void NotNullWhenFalse_EqualsTrue_InErrorInvocation()
+        {
+            CSharpCompilation c = CreateCompilation(new[] { @"
+using System.Runtime.CompilerServices;
+public class C
+{
+    public void Main(string? s)
+    {
+        if (Missing(MyIsNullOrEmpty(s)))
```

Comment:
>Missing(MyIsNullOrEmpty(s)) [](start = 12, length = 27)

Does `[NotNullWhenFalse]` need to be applied to the outermost method to be observable here? For instance, `if (MyIsNullOrEmpty(s)) { ... }` with `static bool MyIsNullOrEmpty([NotNullWhenFalse] Missing? m)`.

Reply:
I meant to validate that the attribute on the nested invocation doesn't accidentally have an effect when the outer invocation is bad. 
The bad invocation still properly collapses the WhenTrue and WhenFalse states, so that conditional attribute has no effect.